src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/ComponentAccessibilityCodeActionProvider.cs

```diff
@@ -0,0 +1,219 @@
+                var actionParams = new AddUsingsCodeActionParams
+                {
+                    Uri = context.Request.TextDocument.Uri,
+                    Namespace = namespaceName,
+                };
+                var data = JObject.FromObject(actionParams);
+
+                var resolutionParams = new RazorCodeActionResolutionParams
+                {
+                    Action = LanguageServerConstants.CodeActions.AddUsing,
+                    Data = data,
+                };
+                var serializedParams = JToken.FromObject(resolutionParams);
+                var arguments = new JArray(serializedParams);
+
+                // Insert @using
+                container.Add(new CommandOrCodeAction(new Command
+                {
+                    Title = $"@using {namespaceName}",
+                    Name = "razor/runCodeAction",
+                    Arguments = arguments,
+                }));
+
+                // Fully qualify
+                container.Add(new CommandOrCodeAction(new CodeAction
+                {
+                    Title = $"{tagHelperPair.Short.Name}",
+                    Edit = CreateRenameTagEdit(context, startTag, tagHelperPair.Short.Name),
+                }));
+            }
+        }
+
+        private static WorkspaceEdit CreateRenameTagEdit(RazorCodeActionContext context, MarkupStartTagSyntax startTag, string newTagName)
+        {
+            var changes = new List<TextEdit>
+            {
+                new TextEdit
+                {
+                    Range = startTag.Name.GetRange(context.CodeDocument.Source),
+                    NewText = newTagName,
+                },
+            };
+            var endTag = ((MarkupElementSyntax)startTag.Parent).EndTag;
+            if (endTag != null)
+            {
+                changes.Add(new TextEdit
+                {
+                    Range = endTag.Name.GetRange(context.CodeDocument.Source),
+                    NewText = newTagName,
+                });
+            }
+            return new WorkspaceEdit
+            {
+                Changes = new Dictionary<Uri, IEnumerable<TextEdit>> {
+                    [context.Request.TextDocument.Uri] = changes,
+                }
+            };
+        }
+
+        private sealed class TagHelperPair
+        {
+            public TagHelperDescriptor Short = null;
+            public TagHelperDescriptor FullyQualified = null;
+        }
+
+        private bool IsTagUnknown(MarkupStartTagSyntax startTag, RazorCodeActionContext context)
+        {
+            foreach (var diagnostic in context.CodeDocument.GetCSharpDocument().Diagnostics)
+            {
+                if (!(diagnostic.Span.AbsoluteIndex > startTag.Span.End || startTag.Span.Start > diagnostic.Span.AbsoluteIndex + diagnostic.Span.Length))
+                {
+                    if (diagnostic.Id == "RZ10012")
```

Comment:
`RZ10012` static readonly. 

Is this the component name starting with lower case?